detox/e2e/support/server_api/system.js

```diff
@@ -56,17 +61,115 @@ export const apiUpdateConfig = async (newConfig = {}) => {
     }
 };
 
-const getDefaultConfig = () => {
+function getDefaultConfig() {
     const fromEnv = {
+        LdapSettings: {
+            LdapServer: testConfig.ldapServer,
+            LdapPort: testConfig.ldapPort,
+        },
         ServiceSettings: {SiteURL: testConfig.siteUrl},
     };
 
     return merge(defaultServerConfig, fromEnv);
+}
+
+/**
+ * Get client license.
+ * See https://api.mattermost.com/#tag/system/paths/~1license~1client/get
+ * @return {Object} returns {license} on success or {error, status} on error
+ */
+export const apiGetClientLicense = async () => {
+    try {
+        const response = await client.get('/api/v4/license/client?format=old');
+
+        return {license: response.data};
+    } catch (err) {
+        return getResponseFromError(err);
+    }
+};
+
+/**
+ * Require server license to successfully continue.
+ * @return {Object} returns {license} on success or fail when no license
+ */
+export const apiRequireLicense = async () => {
+    const {license} = await getClientLicense();
+
+    if (license.IsLicensed !== 'true') {
+        console.error('Server has no Enterprise license.');
+    }
+    jestExpect(license.IsLicensed).toEqual('true');
+
+    return {license};
+};
+
+/**
+ * Require server license with specific feature to successfully continue.
+ * @param {string} key - feature, e.g. LDAP
+ * @return {Object} returns {license} on success or fail when no license or no license to specific feature.
+ */
+export const apiRequireLicenseForFeature = async (key = '') => {
+    const {license} = await getClientLicense();
+
+    if (license.IsLicensed !== 'true') {
+        console.error('Server has no Enterprise license.');
+    }
+    jestExpect(license.IsLicensed).toEqual('true');
+
+    let hasLicenseKey = false;
+    for (const [k, v] of Object.entries(license)) {
+        if (k === key && v === 'true') {
+            hasLicenseKey = true;
+            break;
+        }
+    }
+
+    if (!hasLicenseKey) {
+        console.error(`Server has no license for "${key}" feature.`);
+    }
+    jestExpect(hasLicenseKey).toEqual(true);
+
+    return {license};
+};
+
+/**
+ * Upload server license.
+ * @param {string} absFilePath - absolute path of a license file
+ */
+export const apiUploadLicense = async (absFilePath) => {
+    const response = await apiUploadFile('license', absFilePath, {url: '/api/v4/license', method: 'POST'});
+
+    return response;
 };
 
+/**
+ * Get client license.
+ * If no license, try to upload if license file is available at "/support/fixtures/mattermost-license.txt".
+ */
+async function getClientLicense() {
+    const {license} = await apiGetClientLicense();
+    if (license.IsLicensed === 'true') {
+        return {license};
+    }
+
+    const absPath = path.resolve(__dirname, '../../support/fixtures/mattermost-license.txt');
```

Comment:
how are you going to ensure this file is available in the CI?

Reply:
One option is to have a prior step that adds a file from the environment variable (from context of CircleCI), like
```
echo $EE_LICENSE > detox/e2e/support/fixtures/mattermost-license.txt
```